File: GoSIS/models/personal.go
package models

type Personal struct {
	SQLEmployeeId        int64  `json:"SQL_Employee_ID"`
	FirstName            string `json:"First_Name"`
	LastName             string `json:"Last_Name"`
	MiddleInitial        string `json:"Middle_Initial"`
	Address1             string `json:"Address1"`
	Address2             string `json:"Address2"`
	City                 string `json:"City"`
	State                string `json:"State"`
	Zip                  int64  `json:"Zip"`
	Email                string `json:"Email"`
	PhoneNumber          string `json:"PhoneNumber"`
	SocialSecurityNumber string `json:"SocialSecurityNumber"`
	DriversLicense       string `json:"DriversLicense"`
	MaritalStatus        string `json:"Marital_Status"`
	Gender               bool   `json:"Gender"`
	ShareholderStatus    bool   `json:"Shareholder_Status"`
	BenefitPlans         int64  `json:"Benefit_Plans"`
	Ethnicity            string `json:"Ethnicity"`
}

File: GoSIS/faker-api/personal.py
import requests
import json
from faker import Faker

fake = Faker('vi_VN')  # Vietnamese locale

url = "http://localhost:8080/personal"
headers = {'Content-Type': 'application/json'}

n = int(input("Nhập số lượng request : "))

count = 350
for i in range(n):
    payload = json.dumps({
        "SQL_Employee_ID": count,
        "First_Name": fake.first_name(),
        "Last_Name": fake.last_name(),
        "Middle_Initial": fake.random_letter(),
        "Address1": fake.street_address(),
        "Address2": fake.secondary_address(),
        "City": fake.city(),
        "State": fake.state(),
        "Zip": int(fake.zipcode()),
        "Email": fake.email(),
        "PhoneNumber": fake.phone_number(),
        "SocialSecurityNumber": fake.ssn(),
        "DriversLicense": fake.license_plate(),
        "Marital_Status": fake.random_element(elements=("Single", "Married", "Divorced")),
        "Gender": fake.random_element(elements=(True, False)),
        "Shareholder_Status": fake.random_element(elements=(True, False)),
        "Benefit_Plans": fake.random_int(min=1, max=3),
        "Ethnicity": "Kinh"
    })

    count += 1

    response = requests.request("POST", url, headers=headers, data=payload)

    if response.status_code == 200:
        print(f'POST request {i+1} successful')
    else:
        print(f'Error on POST request {i+1}: {response.text}')
